File: Programmers/level2/rotateParen/rotateParen.py
# https://programmers.co.kr/learn/courses/30/lessons/76502

def solution(s):
    answer = 0
    cnt = len(s)
    while cnt > 0:
        if check(s):
            answer += 1
        s = s[1:] + s[0]
        cnt -= 1
    return answer


def check(c):
    stack = []
    strObj = {
        '(': ')',
        '{': '}',
        '[': ']',
    }
    for char in c:
        if char == '[' or char == '(' or char == '{':
            stack.append(char)
        elif char == ']' or char == ')' or char == '}':
            if not stack:
                return False
            recent = stack.pop()
            if char != strObj[recent]:
                return False
    if len(stack) != 0:
        return False
    else:
        return True

File: Programmers/level2/rotateParen/rotateParen.js
// https://programmers.co.kr/learn/courses/30/lessons/76502?language=javascript

export const solution = (s) => {
    let answer = 0;
    let cnt = s.length;
    while(cnt > 0){
        if(check(s)){
            answer += 1;
        };
        s = s.slice(1) + s.slice(0,1);
        cnt -= 1;
    }
    return answer;
}

const check = (input) => {
  let stackArr = [];
  let strObj = {
    '(': ')',
    '{': '}',
    '[': ']',
  };

  for (let i = 0; i < input.length; i++) {
    if (input[i] === '(' || input[i] === '{' || input[i] === '[') {
      stackArr.push(input[i]);
    } else if (input[i] === ')' || input[i] === '}' || input[i] === ']') {
      let resent = stackArr.pop();
      if (input[i] !== strObj[resent]) {
        return false;
      }
    }
  }

  if (stackArr.length !== 0) {
    return false;
  } else {
    return true;
  }
};